inc/utils.php

```diff
@@ -122,6 +123,63 @@ function query( array $query, array $params = [] ) : ?array {
 	return $result;
 }
 
+/**
+ * Delete old Analytics indexes in Elasticsearch.
+ *
+ * @return array|null
+ */
+function delete_old_indexes() : ?array {
+	// Index age in days.
+	$duration = get_config()['modules']['analytics']['max_index_age'] ?? 7;
+
+	// Get index name by date. 
+	$date = new \DateTime();
+	$date->sub( new \DateInterval( 'P'. $duration . 'D' ) );
+	$index_date = ( $date->format('Y-m-d') ); 
+	$index = 'analytics-' .  $index_date;
+
+	// Get URL.
+	$url = add_query_arg( [], get_elasticsearch_url() . '/' . $index . '/' );
+
+	$response = wp_remote_request( $url, [
+		'method' => 'DELETE',
+	] );
+
+	// Check for failures.
+	if ( wp_remote_retrieve_response_code( $response ) !== 200 || is_wp_error( $response ) ) {
+		if ( is_wp_error( $response ) ) {
+			trigger_error( sprintf(
+				"Analytics: ES index deletion failed: %s",
+				$response->get_error_message()
+			), E_USER_WARNING );
+		} else {
+			trigger_error( sprintf(
+				"Analytics: ES index deletion failed:\n%s\n%s",
+				json_encode( $index ),
```

Comment:
I think it'll be easier to read the logs if this uses `JSON_PRETTY_PRINT` and a few other useful flags for output eg:
```suggestion
				json_encode( $index, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE ),
```